packages/webview_flutter/webview_flutter_wkwebview/test/src/web_kit_webview_widget_test.dart

```diff
@@ -205,6 +259,79 @@ void main() {
       });
 
       group('$WebSettings', () {
+        testWidgets('javascriptMode', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                javascriptMode: JavascriptMode.unrestricted,
+                hasNavigationDelegate: false,
+              ),
+            ),
+          );
+
+          verify(mockPreferences.setJavaScriptEnabled(true));
+        });
+
+        testWidgets('hasNavigationDelegate', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                hasNavigationDelegate: true,
+              ),
+            ),
+          );
+
+          verify(mockNavigationDelegate
+              .setDecidePolicyForNavigationAction(argThat(isNotNull)));
+        });
+
+        testWidgets('userAgent', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.of('myUserAgent'),
+                hasNavigationDelegate: false,
+              ),
+            ),
+          );
+
+          verify(mockWebView.setCustomUserAgent('myUserAgent'));
+        });
+
+        testWidgets('zoomEnabled', (WidgetTester tester) async {
```

Comment:
This seems like a confusing name for testing zoomEnabled=false.

Reply:
updated name to `zoomEnabled is false`